core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/util/JsonUtil.java

```diff
@@ -39,20 +38,23 @@ public class JsonUtil {
 
     private JsonUtil() {}
 
-    public static JSONArray objectsToJsonArray(Collection<?> objects) {
-        JSONArray jsonArray = new JSONArray();
+    public static JsonArray objectsToJsonArray(Collection<?> objects) {
```

Comment:
To replace this whole method:

```suggestion
    public static ArrayNode objectsToJsonArray(Collection<?> objects) {
        ArrayNode arrayNode = OBJECT_MAPPER.createArrayNode();
        objects.forEach(object -> {
            try {
                JsonNode jsonNode = OBJECT_MAPPER.valueToTree(object);
                arrayNode.add(jsonNode);
            } catch (IllegalArgumentException e) {
                LOG.error("Failed to convert gridResources to JSON", e);
            }
        });
        return arrayNode;
    }
```